nvflare/private/fed/client/communicator.py

```diff
@@ -391,7 +391,7 @@ def quit_remote(self, servers, task_name, token, ssid, fl_ctx: FLContext):
                     time.sleep(3)
         return server_message
 
-    def send_heartbeat(self, servers, task_name, token, ssid, client_name):
+    def send_heartbeat(self, servers, task_name, token, ssid, client_name, engine: ClientEngine):
```

Comment:
I was thinking not just to the newly added arguments, but whole method such as
servers: list, task_name: str, token : str, ssid: str, client_name :string 
and return type

Reply:
I guess this probably doesn't just apply this methods, but all others, you can choose to do it later if you wish